expertsystem/state/conservation_rules.py

```diff
@@ -571,8 +562,45 @@ def check_particles_identical(particles):
                 return False
         return True
 
+    if _check_particles_identical(outgoing_part_qns):
+        parity_label = StateQuantumNumberNames.Parity
+
+        if is_boson(outgoing_part_qns[0]):
+            # we have a boson, check if parity of mother is even
+            parity = ingoing_part_qns[0][parity_label]
+            if parity == -1:
+                # if its odd then return False
+                return False
+        else:
+            # its fermion
+            parity = ingoing_part_qns[0][parity_label]
+            if parity == 1:
+                return False
+
+    return True
+
+
+def _is_clebsch_gordan_coefficient_zero(spin1, spin2, spin_coupled):
+    m_1 = spin1.projection
+    j_1 = spin1.magnitude
+    m_2 = spin2.projection
+    j_2 = spin2.magnitude
+    proj = spin_coupled.projection
+    mag = spin_coupled.magnitude
+    is_zero = False
+    if (j_1 == j_2 and m_1 == m_2) or (m_1 == 0.0 and m_2 == 0.0):
+        if abs(mag - j_1 - j_2) % 2 == 1:
+            is_zero = True
+    elif j_1 == mag and m_1 == -proj:
+        if abs(j_2 - j_1 - mag) % 2 == 1:
+            is_zero = True
+    elif j_2 == mag and m_2 == -proj:
+        if abs(j_1 - j_2 - mag) % 2 == 1:
+            is_zero = True
+    return is_zero
+
 
-class SpinConservation(AbstractRule):
+class SpinConservation(Rule):
```

Comment:
This is just like the `AdditiveQuantumNumberConservation` rule. Its a general class to check "Spin" conservation, so both Spin (+angular momentum) as well as Isospin create an instance of this class. Since I need to access the correct quantum numbers in the actual rule check function, I need some way to keep the state. I don't see how this can be achieved with the additional argument